pymc3/distributions/simulator.py

```
@@ -0,0 +1,66 @@
+import theano
+import theano.tensor as tt
+import numpy as np
+from .distribution import NoDistribution, Distribution
+
+__all__ = ['Simulator']
+
+
+class Simulator(NoDistribution):
+
+    def __init__(self, function, *args, **kwargs):
+
+        self.function = function
+        observed = self.data
+        super(
+            Simulator,
+            self).__init__(
+            shape=np.prod(
+                observed.shape),
+            dtype=observed.dtype,
+            *args,
+            **kwargs)
+
+    def random(self, point=None, size=None):
+        """
+        Draw random values from Simulator
+
+        Parameters
+        ----------
+        point : dict, optional
+            Dict of variable values on which random values are to be
+            conditioned (uses default point if not specified).
+        size : int, optional
+            Desired size of random sample (returns one sample if not
+            specified).
+
+        Returns
+        -------
+        array
+        """
+
+        print('Not implemented yet')
+
```

Fail fast: `raise NotImplementedError('Who is in charge of implementing this?')`

---

The random method of the simulator should be just the `function`, right?